File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int left = 0;
        int right = 1;
        int maxP = 0;

        while (right < prices.length){
            if(prices[left] < prices[right]){
                int profit = prices[right] - prices[left];
                maxP = Math.max(profit,maxP);
            }else{
                left = right;
            }
            right++;
        }

        return maxP;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {

        let left = 0;
        let right = heights.length - 1;
        let maxArea = 0;

        while(left< right){
            let width = (right - left );

            if(heights[left] < heights[right]){
                let height = heights[left]
                let currArea = width * height;

                //need to calculate the max area
                maxArea = Math.max(currArea,maxArea);
                left++;
            }else{
                let height = heights[right]
                let currArea = width * height;

                maxArea = Math.max(currArea,maxArea);
                right--
            }
        }
        return maxArea;
    }
}
